fvm/runtime/reusable_cadence_runtime.go

```diff
@@ -4,13 +4,26 @@ import (
 	"github.com/onflow/cadence"
 	"github.com/onflow/cadence/runtime"
 	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
 	"github.com/onflow/cadence/runtime/sema"
+	"github.com/onflow/cadence/runtime/stdlib"
+
+	"github.com/onflow/flow-go/fvm/errors"
 )
 
 // Note: this is a subset of environment.Environment, redeclared to handle
 // circular dependency.
 type Environment interface {
 	runtime.Interface
+
+	RandomSourceHistory() ([]byte, error)
```

Comment:
Why is `RandomSourceHistory` not part of `runtime.Interface`?

Reply:
Is it because random source history is only available for transaction, but not script?

Reply:
`runtime.Interface` is the cadence interface, but `RandomSourceHistory` is not part of the cadence interface. It only needs to be part of the environment.